sdk/evaluation/azure-ai-evaluation/azure/ai/evaluation/_legacy/prompty/_prompty.py

```diff
@@ -368,15 +373,36 @@ async def _send_with_retries(
         should_retry: bool = True
         retry: int = 0
         delay: Optional[float] = None
+        # Make a mutable copy of the parameters
+        mutable_params = dict(params)
 
         while should_retry:
             try:
                 if delay:
                     await asyncio.sleep(delay)
 
-                response = await client.chat.completions.create(**params)
+                response = await client.chat.completions.create(**mutable_params)
                 return response
             except OpenAIError as error:
+                error_message = str(error)
+                BAD_REQUEST_ERROR_CODE = "400"
+
+                # Check for BadRequestError with unsupported parameter message
+                if (not self._is_reasoning_model and
```

Comment:
If there is another series of model come along which take diferent params this solution will render ineffective.